parsers/MD.py

```diff
@@ -1,103 +1,96 @@
-        "currency": "MDL",
-        "price": price,
-        "source": "moldelectrica.md",
-    }
-
-
-def template_consumption_response(
-    zone_key: str, datetime: datetime, consumption
-) -> dict:
-    return {
-        "zoneKey": zone_key,
-        "datetime": datetime,
-        "consumption": consumption,
-        "source": "moldelectrica.md",
-    }
-
-
-def template_production_response(zone_key: str, datetime: datetime, production) -> dict:
-    return {
-        "zoneKey": zone_key,
-        "datetime": datetime,
-        "production": production,
-        "storage": {},
-        "source": "moldelectrica.md",
-    }
-
-
-def template_exchange_response(
-    sorted_zone_keys: str, datetime: datetime, netflow
-) -> dict:
-    return {
-        "sortedZoneKeys": sorted_zone_keys,
-        "datetime": datetime,
-        "netFlow": netflow,
-        "source": "moldelectrica.md",
-    }
-
-
-def get_archive_data(
+ARCHIVE_BASE_URL = "https://moldelectrica.md/utils/archive2.php?id=table&type=array"
+SOURCE = "moldelectrica.md"
+
+
+# Moldoelectrica electricity tariffs as defined by government-agency decisions.
+_MOLDOELECTRICA_NEW_POWER_PRICE_IN_MDL_PER_MW = {
+    # https://www.legis.md/cautare/getResults?doc_id=78826&lang=ro
+    datetime(2000, 4, 1, tzinfo=timezone.utc): 18.8,
+    # https://www.legis.md/cautare/getResults?doc_id=103953&lang=ro
+    datetime(2001, 10, 1, tzinfo=timezone.utc): 28.0,
+    # https://www.legis.md/cautare/getResults?doc_id=40249&lang=ro
+    datetime(2002, 9, 1, tzinfo=timezone.utc): 35.2,
+    # https://www.legis.md/cautare/getResults?doc_id=42701&lang=ro
+    datetime(2005, 9, 1, tzinfo=timezone.utc): 39.3,
+    # https://www.legis.md/cautare/getResults?doc_id=10948&lang=ro
+    datetime(2007, 8, 3, tzinfo=timezone.utc): 51.8,
+    # https://www.legis.md/cautare/getResults?doc_id=40130&lang=ro
+    datetime(2010, 1, 19, tzinfo=timezone.utc): 63.2,
+    # https://www.legis.md/cautare/getResults?doc_id=40589&lang=ro
+    datetime(2012, 5, 11, tzinfo=timezone.utc): 80.2,
+    # https://www.legis.md/cautare/getResults?doc_id=84436&lang=ro
+    datetime(2015, 7, 31, tzinfo=timezone.utc): 145.0,
+    # https://www.legis.md/cautare/getResults?doc_id=134854&lang=ro
+    datetime(2023, 12, 31, tzinfo=timezone.utc): 201.0,
+    # https://www.legis.md/cautare/getResults?doc_id=142391&lang=ro
+    datetime(2024, 3, 21, tzinfo=timezone.utc): 185.0,
+    # TODO(amv213): update as new tariffs get rolled out...
+}
+
+
+class ArchiveDatapoint(NamedTuple):
+    """Datapoint returned by the archive_url with ordered fetchable fields."""
+
+    datetime: datetime
+    consumption: float
+    planned_consumption: float
+    production: float
+    planned_production: float
+    tpp: float  # production from thermal power plants
+    hpp: float  # production from hydro power plants
+    res: float  # production from renewable energy sources
+    exchange_UA_to_MD: float
+    planned_exchange_UA_to_MD: float
```

Comment:
Would it be possible to add exchange forecasts as well? I see the parser fon't contain it right now. Feel free to do that in a seperate PR though.

Reply:
Sure! I can try having a look into that 🙌🏽 I'll do that in a separate PR

Reply:
@VIKTORVAV99 what would be the best way to implement this? Would it be just a matter of adding to the existing values returned by the `fetch_exchange()` function new datapoints with the planned values and "forecasted" source type like in the example [here](https://github.com/electricitymaps/electricitymaps-contrib/wiki/Building-a-new-parser#fetch_exchange_forecast) or would it be a dedicated separate function just returning forecasts? I saw that some other parsers are using something called `fetch_exchange_forecast`

Reply:
Oh yeah it should be a different function. I'll make sure to update the wiki.